mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -3972,20 +4008,35 @@ func expandIPAllocationPolicy(configured interface{}, networkingMode string) (*c
 		return &container.IPAllocationPolicy{
 			UseIpAliases:    false,
 			UseRoutes:       true,
+			StackType:       "IPV4",
 			ForceSendFields: []string{"UseIpAliases"},
 		}, nil
 	}
 
 	config := l[0].(map[string]interface{})
+	stackType := config["stack_type"].(string)
+
+	if stackType == "IPV4_IPV6" {
+		if datapathProvider != "ADVANCED_DATAPATH" {
+			return nil, fmt.Errorf("Dual-stack networking is only available for VPC-native clusters with GKE DataplaneV2 enabled.")
+		}
+		if privateIpv6GoogleAccess == "PRIVATE_IPV6_GOOGLE_ACCESS_TO_GOOGLE" ||  privateIpv6GoogleAccess == "PRIVATE_IPV6_GOOGLE_ACCESS_BIDIRECTIONAL" {
+			return nil, fmt.Errorf("Dual-stack clusters doesn't support Private Google Access over IPv6.")
+		}
+		if !enableL4ilbSubnetting {
+			return nil, fmt.Errorf("For dual-stack clusters, GKE will enable L4 ILB Subnetting automatically. Enable L4 ILB Subnetting to avoid drift.")
+		}
+	}
```

Comment:
Does the API have validations similar to this? We tend to minimize the custom validations on Terraform side and let the API enforce them. The API-side validation can change over time and it will cause provider lag behind the real requirements. Sorry that I forgot to point it out in the last review.

Reply:
Totally agreed. This is very good point. I'll remove the validation on TF side and leave the validation to the API itself